WatchDogEclipsePlugin/WatchDog/src/nl/tudelft/watchdog/logic/interval/IntervalManagerBase.java

```diff
@@ -73,4 +75,21 @@ public IntervalBase getIntervalOfType(IntervalType type) {
 		}
 		return collectedIntervals;
 	}
+
+	/**
+	 * @return An {@link ArrayList} of intervals of the specified Perspective
+	 *         type.
+	 */
+	protected List<IntervalBase> getPerspectiveIntervalsOfType(Perspective type) {
```

Comment:
I am not afraid of performance issues in this case.

It seems more efficient to me to reverse roles here, and I'm having a go at it, ok?